File: library-base/src/main/java/com/my/library_base/utils/MMKVUtils.java
package com.my.library_base.utils;

import com.my.library_base.constants.Constants;
import com.tencent.mmkv.MMKV;

public class MMKVUtils {
    private static MMKVUtils mmkvUtils;
    private MMKV mmkv;

    private MMKVUtils(String id) {
        mmkv = MMKV.mmkvWithID(id);
    }

    public static MMKVUtils getInstance() {
        if (null == mmkvUtils) {
            mmkvUtils = new MMKVUtils(Constants.MMKV_ID);
        }
        return mmkvUtils;
    }

    public void setValue(String key, String value) {
        mmkv.encode(key, value);
    }

    public void setValue(String key, boolean value) {
        mmkv.encode(key, value);
    }

    public void setValue(String key, int value) {
        mmkv.encode(key, value);
    }

    public String getStringValue(String key) {
        return mmkv.decodeString(key, "");
    }

    public boolean getBooleanValue(String key) {
        return mmkv.decodeBool(key, false);
    }

    public int getIntValue(String key) {
        return mmkv.decodeInt(key, -1);
    }

    public void remove(String key) {
        mmkv.remove(key);
    }

    public String[] getAllKey() {
        return mmkv.allKeys();
    }

    public boolean contain(String key) {
        return mmkv.contains(key);
    }
}

File: library-base/src/main/java/com/my/library_base/utils/MMKVUtils.kt
package com.my.library_base.utils

import com.tencent.mmkv.MMKV

class MMKVUtils private constructor(id: String) {
    private val mmkv: MMKV?
    fun setValue(key: String?, value: String?) {
        mmkv!!.encode(key, value)
    }

    fun setValue(key: String?, value: Boolean) {
        mmkv!!.encode(key, value)
    }

    fun setValue(key: String?, value: Int) {
        mmkv!!.encode(key, value)
    }

    fun getStringValue(key: String?): String? {
        return mmkv!!.decodeString(key)
    }

    fun getBooleanValue(key: String?): Boolean {
        return mmkv!!.decodeBool(key)
    }

    fun getIntValue(key: String?): Int {
        return mmkv!!.decodeInt(key)
    }

    fun remove(key: String?) {
        mmkv!!.remove(key)
    }

    val allKey: Array<String>?
        get() = mmkv!!.allKeys()

    fun contain(key: String?): Boolean {
        return mmkv!!.contains(key)
    }

    companion object {
        private var mmkvUtils: MMKVUtils? = null
        fun getInstance(id: String): MMKVUtils? {
            if (null == mmkvUtils) {
                mmkvUtils = MMKVUtils(id)
            }
            return mmkvUtils
        }
    }

    init {
        mmkv = MMKV.mmkvWithID(id)
    }
}